pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -82,18 +91,64 @@ type CertificateManager interface {
 	AddToStatusManager(manager status.StatusManager, namespace string)
 	// KeyPair Returns the CA KeyPairInterface, so it can be rendered in the operator namespace.
 	KeyPair() certificatemanagement.KeyPairInterface
+	// Loads an existing trusted bundle to pass to render.
+	LoadTrustedBundle(context.Context, client.Client, string) (certificatemanagement.TrustedBundleRO, error)
+}
+
+type Option func(cm *certificateManager) error
+
+func AllowCACreation() Option {
```

Comment:
I suppose it might be cleaner to have a "CreateCA" function that is distinct from this, and just always prevent creation of the CA from this function. That's probably better.

Reply:
I think I will try this as a separate PR from other feedback though - this one is a bit gnarly to unpick and this code scares me a bit.